src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/ReferenceTransactionNumber.java

```diff
@@ -1,29 +1,56 @@
 package uk.nhs.digital.nhsconnect.nhais.model.edifact;
 
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
 import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
 
-public class ReferenceTransactionNumber extends Reference {
+import org.springframework.util.StringUtils;
 
-    public ReferenceTransactionNumber() {
-        super("TN", "");
+@RequiredArgsConstructor @NoArgsConstructor
+public class ReferenceTransactionNumber extends Segment {
+
+    public static final String KEY = "RFF";
+    public static final String QUALIFIER = "TN";
+    public static final String KEY_QUALIFIER = KEY + "+" + QUALIFIER;
+    private @NonNull String transactionNumber;
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        return QUALIFIER + ":" + transactionNumber;
+    }
+
+    @Override
+    public void preValidate() throws EdifactValidationException {
+        //NOP
     }
 
     public Long getTransactionNumber() {
-        return Long.decode(getReference());
+        return Long.decode(transactionNumber);
     }
 
     public ReferenceTransactionNumber setTransactionNumber(Long transactionNumber) {
-        this.setReference(Long.toString(transactionNumber));
+        this.transactionNumber = Long.toString(transactionNumber);
         return this;
     }
 
     @Override
     protected void validateStateful() throws EdifactValidationException {
-        if (getReference() == null) {
-            throw new EdifactValidationException(getKey() + ": Attribute qualifier is required");
+        if (StringUtils.isEmpty(transactionNumber)) {
```

Comment:
transactionNumber should be a number type so that it can be validated properly.Only whole numbers in the range [1, 9999999] are valid transaction numbers.

Reply:
Transaction numbers are not zero-padded when they are written to edifact

Reply:
I think this is calling StringUtils.isEmpty on type Long now?

Reply:
I'll validate that in my PR